be/src/storage/txn_manager.cpp

```diff
@@ -363,17 +363,22 @@ Status TxnManager::persist_tablet_related_txns(const std::vector<TabletSharedPtr
         persisted.insert(path);
     }
 
+    // using BThreadCountDownLatch to make sure it wouldn't block the brpc worker.
+    BThreadCountDownLatch bthread_latch(to_flush_tablet.size());
     auto token = _flush_thread_pool->new_token(ThreadPool::ExecutionMode::CONCURRENT);
     std::vector<std::pair<Status, int64_t>> pair_vec(to_flush_tablet.size());
     int i = 0;
     for (auto& tablet : to_flush_tablet) {
         auto dir = tablet->data_dir();
-        token->submit_func([&pair_vec, dir, i]() { pair_vec[i].first = dir->get_meta()->flushWAL(); });
+        token->submit_func([&pair_vec, &bthread_latch, dir, i]() {
```

Comment:
Should check and handle the return value

Reply:
Done